exporter/exporterhelper/internal/bounded_memory_queue_test.go

```diff
@@ -150,6 +151,62 @@ func TestShutdownWhileNotEmpty(t *testing.T) {
 	assert.Equal(t, 0, q.Size())
 }
 
+func Benchmark_QueueUsage_10000_1_50000(b *testing.B) {
+	queueUsage(b, 10000, 1, 50000)
+}
+
+func Benchmark_QueueUsage_10000_2_50000(b *testing.B) {
+	queueUsage(b, 10000, 2, 50000)
+}
+func Benchmark_QueueUsage_10000_5_50000(b *testing.B) {
+	queueUsage(b, 10000, 5, 50000)
+}
+func Benchmark_QueueUsage_10000_10_50000(b *testing.B) {
+	queueUsage(b, 10000, 10, 50000)
+}
+
+func Benchmark_QueueUsage_50000_1_50000(b *testing.B) {
+	queueUsage(b, 50000, 1, 50000)
+}
+
+func Benchmark_QueueUsage_50000_2_50000(b *testing.B) {
+	queueUsage(b, 50000, 2, 50000)
+}
+func Benchmark_QueueUsage_50000_5_50000(b *testing.B) {
+	queueUsage(b, 50000, 5, 50000)
+}
+func Benchmark_QueueUsage_50000_10_50000(b *testing.B) {
+	queueUsage(b, 50000, 10, 50000)
+}
+
+func Benchmark_QueueUsage_10000_1_250000(b *testing.B) {
+	queueUsage(b, 10000, 1, 250000)
+}
+
+func Benchmark_QueueUsage_10000_2_250000(b *testing.B) {
+	queueUsage(b, 10000, 2, 250000)
+}
+func Benchmark_QueueUsage_10000_5_250000(b *testing.B) {
+	queueUsage(b, 10000, 5, 250000)
+}
+func Benchmark_QueueUsage_10000_10_250000(b *testing.B) {
+	queueUsage(b, 10000, 10, 250000)
+}
+
+func queueUsage(b *testing.B, capacity int, numConsumers int, numberOfItems int) {
+	for i := 0; i < b.N; i++ {
```

Comment:
Please add memory allocation reporting

Reply:
Sorry, I'm not sure how that is best done - googling gives me weird results. Any example?

Reply:
Just add `b.ReportAllocs()`

Reply:
Done, thanks!